Source/cryptalgo/SecureSocketPort.h

```diff
@@ -72,19 +72,26 @@ namespace Crypto {
             };
 
         public:
+            // Type of underlying socket
```

Comment:
This enum, together with the enum class context_t split up into server and client do not add any functionality. They are currently used to "determine" the initial state of the _handshaking in the handler but the handler, in the initialize will override the set state, by definition, through the IsOpen() state, which indeed is the only valid way to determine if you are a server, so why all this additional code?
I suggest setting the initial satet at ERROR and if during normal runtime you use this state and it is still ERROR, assert where you would not expect it..

Reply:
Good observation. It was not the intended goal to have it overwritten.